### ISSUE TITLE
Сan't transfer learning

### ISSUE THREAD

**Issue by AlfaLegion** [Microsoft/CNTK#3610]
I'm trying to make  transfer learning
following the example from the lesson:
[](https://cntk.ai/pythondocs/CNTK_301_Image_Recognition_with_Deep_Transfer_Learning.html)

I have dataset images with shape(3x53x128) CHW notation
I'm change shape (3,224,224) to shape (3,53,128) and run 
```
base_model = {
    'model_file': base_model_file,
    'feature_node_name': 'features',
    'last_hidden_node_name': 'z.x',
    # Channel Depth x Height x Width
    'image_dims': (3, 53, 128)   #before 'image_dims': (3,224,224)
}
```
I get a dimension error: Convolution operation requires that kernel dim 3 <= input dim 2.
```
def create_model(...):
     ...
     cloned_out = cloned_layers(feat_norm)   <<<-----Error
     ....
    return z
```

Help me please, I do not know what the problem is.
(sorry for my english)

**Comment by delzac** [comment]
Typically, in a computer vision deep learning model, both convolutional, pooling and dense layers are used. While convolutional are input size agnostic, the same is not true for dense layers which are typically found in the last layer in the model and only takes in a fix dimension. Changing the input size changes the input dimension to the dense layer and breaks it.

The network you are using contains dense layer and hence you cannot just change the input dimensions. I suggest you resize you image instead.

**Comment by AlfaLegion** [comment]
This network has one fully connected layer which is located at the end - layer classification. I'm cut before classification layer. Name cut layer: "z.x.x.x.p".  Here I get a dimension error. The exact location where the error comes from is a convolution layer named "_z.x.x.x.b.x.c" .I do not understand why there is an error

**Comment by delzac** [comment]
As you convolute, the output will be smaller than the input without padding. That's what happened for your case.

**Comment by delzac** [comment]
The image is progressively becoming smaller due to convolution and pooling. Typically, architectures are designed such that the last pooling layer pools a 7px by 7px feature map.

**Comment by AlfaLegion** [comment]
Checked at each level the dimensions of the output tensors. The dimension of the entry convolution _z.xxxbxc 512x2x6. Why is there no such problem in the PyTorch? Is it related to dynamic graph and broadcasting input tensor?

**Comment by delzac** [comment]
As the error suggest, you can't do convolution when your kernel is 3x3 and you feature map is 2x6. 3 is larger than 2, there's no way to convolute without padding.

It has nothing to do with dynamic graph or broadcasting. You should check the padding settings in pytorch. It shouldn't be different in pytorch, tensorflow or any other framework.

**Comment by AlfaLegion** [comment]
in the pyTorch, I just change output dimanshion in the classification layer

**Comment by AlfaLegion** [comment]
This is list output shape each layers PyTorch for input image (3,53,128):
```
----------------------------------------------------------------
        Layer (type)               Output Shape         Param #
================================================================
            Conv2d-1           [-1, 64, 27, 64]           9,408
       BatchNorm2d-2           [-1, 64, 27, 64]             128
              ReLU-3           [-1, 64, 27, 64]               0
         MaxPool2d-4           [-1, 64, 14, 32]               0
            Conv2d-5           [-1, 64, 14, 32]          36,864
       BatchNorm2d-6           [-1, 64, 14, 32]             128
              ReLU-7           [-1, 64, 14, 32]               0
            Conv2d-8           [-1, 64, 14, 32]          36,864
       BatchNorm2d-9           [-1, 64, 14, 32]             128
             ReLU-10           [-1, 64, 14, 32]               0
       BasicBlock-11           [-1, 64, 14, 32]               0
           Conv2d-12           [-1, 64, 14, 32]          36,864
      BatchNorm2d-13           [-1, 64, 14, 32]             128
             ReLU-14           [-1, 64, 14, 32]               0
           Conv2d-15           [-1, 64, 14, 32]          36,864
      BatchNorm2d-16           [-1, 64, 14, 32]             128
             ReLU-17           [-1, 64, 14, 32]               0
       BasicBlock-18           [-1, 64, 14, 32]               0
           Conv2d-19           [-1, 128, 7, 16]          73,728
      BatchNorm2d-20           [-1, 128, 7, 16]             256
             ReLU-21           [-1, 128, 7, 16]               0
           Conv2d-22           [-1, 128, 7, 16]         147,456
      BatchNorm2d-23           [-1, 128, 7, 16]             256
           Conv2d-24           [-1, 128, 7, 16]           8,192
      BatchNorm2d-25           [-1, 128, 7, 16]             256
             ReLU-26           [-1, 128, 7, 16]               0
       BasicBlock-27           [-1, 128, 7, 16]               0
           Conv2d-28           [-1, 128, 7, 16]         147,456
      BatchNorm2d-29           [-1, 128, 7, 16]             256
             ReLU-30           [-1, 128, 7, 16]               0
           Conv2d-31           [-1, 128, 7, 16]         147,456
      BatchNorm2d-32           [-1, 128, 7, 16]             256
             ReLU-33           [-1, 128, 7, 16]               0
       BasicBlock-34           [-1, 128, 7, 16]               0
           Conv2d-35            [-1, 256, 4, 8]         294,912
      BatchNorm2d-36            [-1, 256, 4, 8]             512
             ReLU-37            [-1, 256, 4, 8]               0
           Conv2d-38            [-1, 256, 4, 8]         589,824
      BatchNorm2d-39            [-1, 256, 4, 8]             512
           Conv2d-40            [-1, 256, 4, 8]          32,768
      BatchNorm2d-41            [-1, 256, 4, 8]             512
             ReLU-42            [-1, 256, 4, 8]               0
       BasicBlock-43            [-1, 256, 4, 8]               0
           Conv2d-44            [-1, 256, 4, 8]         589,824
      BatchNorm2d-45            [-1, 256, 4, 8]             512
             ReLU-46            [-1, 256, 4, 8]               0
           Conv2d-47            [-1, 256, 4, 8]         589,824
      BatchNorm2d-48            [-1, 256, 4, 8]             512
             ReLU-49            [-1, 256, 4, 8]               0
       BasicBlock-50            [-1, 256, 4, 8]               0
           Conv2d-51            [-1, 512, 2, 4]       1,179,648
      BatchNorm2d-52            [-1, 512, 2, 4]           1,024
             ReLU-53            [-1, 512, 2, 4]               0
           Conv2d-54            [-1, 512, 2, 4]       2,359,296
      BatchNorm2d-55            [-1, 512, 2, 4]           1,024
           Conv2d-56            [-1, 512, 2, 4]         131,072
      BatchNorm2d-57            [-1, 512, 2, 4]           1,024
             ReLU-58            [-1, 512, 2, 4]               0
       BasicBlock-59            [-1, 512, 2, 4]               0
           Conv2d-60            [-1, 512, 2, 4]       2,359,296
      BatchNorm2d-61            [-1, 512, 2, 4]           1,024
             ReLU-62            [-1, 512, 2, 4]               0
           Conv2d-63            [-1, 512, 2, 4]       2,359,296
      BatchNorm2d-64            [-1, 512, 2, 4]           1,024
             ReLU-65            [-1, 512, 2, 4]               0
       BasicBlock-66            [-1, 512, 2, 4]               0
AdaptiveAvgPool2d-67            [-1, 512, 1, 1]               0
           Linear-68                    [-1, 2]           1,026
================================================================
Total params: 11,177,538
Trainable params: 11,177,538
Non-trainable params: 0
----------------------------------------------------------------
Input size (MB): 0.08
Forward/backward pass size (MB): 9.07
Params size (MB): 42.64
Estimated Total Size (MB): 51.78](url)
```

I am not getting dimension errors

**Comment by delzac** [comment]
It seems like you are padding in pytorch but not in cntk. Set pad=True in cntk and it will work just like what you did in pytorch. Pad is set as false by default in cntk conv layer.

**Comment by AlfaLegion** [comment]
Can I do this already in a pre-trained model?

**Comment by delzac** [comment]
The padding is specificied by the imported model. If you insist on not resizing your image, then the only option is to create a Convolution layer and initialise the weights from the pre-trained model. It will be a bit more work, but then you get to control the padding.

**Comment by AlfaLegion** [comment]
Thanks for the help))
